logic/peers.js

```diff
@@ -0,0 +1,166 @@
+'use strict';
+
+var _ = require('lodash');
+var async = require('async');
+var Peer = require('../logic/peer.js');
+var schema = require('../schema/peers.js');
+
+// Private fields
+var __private = {};
+var self;
+var modules;
+var library;
+
+// Constructor
+function Peers (cb) {
+	self = this;
+	__private.peers = {};
+	return setImmediate(cb, null, this);
+}
+
+Peers.prototype.create = function (peer) {
+	if (!(peer instanceof Peer)) {
+		return new Peer(peer);
+	} else {
+		return peer;
+	}
+};
+
+Peers.prototype.exists = function (peer) {
+	peer = self.create(peer);
+	return !!__private.peers[peer.string];
+};
+
+Peers.prototype.get = function (peer) {
+	if (typeof peer === 'string') {
```

Comment:
maybe create some function  'isAddress(peer)' with regexp that will check if it's address or not.

Or maybe split this into 2 functions get and getByAddress